base/threads.jl

```diff
@@ -11,25 +11,4 @@ include("threadingconstructs.jl")
 include("atomics.jl")
 include("locks-mt.jl")
 
-
-"""
-    resize_nthreads!(A, copyvalue=A[1])
-
-Resize the array `A` to length [`nthreads()`](@ref).   Any new
-elements that are allocated are initialized to `deepcopy(copyvalue)`,
-where `copyvalue` defaults to `A[1]`.
-
-This is typically used to allocate per-thread variables, and
-should be called in `__init__` if `A` is a global constant.
-"""
-function resize_nthreads!(A::AbstractVector, copyvalue=A[1])
-    nthr = nthreads()
-    nold = length(A)
-    resize!(A, nthr)
-    for i = nold+1:nthr
-        A[i] = deepcopy(copyvalue)
-    end
-    return A
-end
-
```

Comment:
Would a more object-oriented API be good for replacing this?
```julia
struct ThreadLocalData
    f::Function
    l::Lock
    d::Vector{Any}
end
Base.getindex(tld::ThreadLocalData) = ...
```

e.g. https://github.com/JuliaLang/julia/issues/14135

Reply:
ref https://juliaconcurrent.github.io/ConcurrentUtils.jl/dev/#ConcurrentUtils.ThreadLocalStorage